libnmstate/nispor/macvlan.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import MacVlan
+
+from .base_iface import NisporPluginBaseIface
+
+
+MACVLAN_FLAG_NOPROMISC = 1
+
+
+class NisporPluginMacVlanIface(NisporPluginBaseIface):
+    @property
+    def type(self):
+        return InterfaceType.MAC_VLAN
+
+    def to_dict(self):
+        info = super().to_dict()
+        info[MacVlan.CONFIG_SUBTREE] = {
+            MacVlan.BASE_IFACE: self._np_iface.base_iface,
+            MacVlan.MODE: self._np_iface.mode,
```

Comment:
The nispor might return `Other(u32)` when new mode pop up.
Please map it to `MacVlan.Mode.UNKNOWN`.

The best approach would be create a 1:1 map and use `MacVlan.Mode.UNKNOWN` as fallback.
So that we don't get impacted when nispor add new mode or got mode don't know.

Reply:
Done, thanks!